### PR TITLE
 Bug 2023228: Remove Tech preview badge for the triggers component for triggers GA …

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/OCPBUGSM-37184

**Analysis / Root cause**: 

OSP 1.6 contains Triggers 0.16 (GA version of triggers components), so we need to remove the Tech preview badges on the OCP UI 4.9.

**Solution Description**: 

Removed the Tech preview badges for the OSP 1.6 and above operators.

**Screen shots / Gifs for design review**: 

Screenshot 2021-11-16 at 2 26 33 PM
**Trigger Template page:**
Screenshot 2021-11-16 at 3 04 32 PM

**Admin -> Triggers Page:**
Screenshot 2021-11-16 at 3 04 43 PM

**EventListenerPage:**
Screenshot 2021-11-16 at 3 04 51 PM
**TriggerBinding Page:**
Screenshot 2021-11-16 at 3 05 11 PM
**ClusterTriggerBinding Page:**
Screenshot 2021-11-16 at 3 05 25 PM

**Unit test coverage report**: 

image

**Test setup:**

Add Trigger to a pipeline and visit all the Trigger components related details pages.

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
cc: @sanketpathak @VeereshAradhya

File: frontend/packages/pipelines-plugin/src/utils/hooks.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+
 export const usePipelineTechPreviewBadge = (namespace: string) => {
   const operator = usePipelineOperatorVersion(namespace);
   if (!operator) return null;
-  const installedGA = isGAVersionInstalled(operator);
-  return installedGA ? null : getBadgeFromType(BadgeType.TECH);
+  return getOSPTechPreviewBadge(isGAVersionInstalled(operator));
+};
+
+export const useTriggersTechPreviewBadge = (namespace: string) => {
```
Comment (abhinandan13jan) [inline frontend/packages/pipelines-plugin/src/utils/hooks.ts:17]: Should we not use a generic `getBadge` function and return badge using `getBadgeFromType`?? WDYT??

Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/utils/hooks.ts:17]: The utility function to return the badges `getOSPTechPreviewBadge` internally uses getBadgeFromType function.

File: frontend/packages/pipelines-plugin/src/components/pipelines/utils/pipeline-operator.ts [hunk 2]
```diff
@@ -44,3 +44,8 @@ export const isGAVersionInstalled = (operator: SemVer): boolean => {
   if (!operator) return false;
   return gte(operator.version, PIPELINE_GA_VERSION);
 };
+
+export const isTriggersGAVersionInstalled = (operator: SemVer): boolean => {
```
Comment (abhinandan13jan) [inline frontend/packages/pipelines-plugin/src/components/pipelines/utils/pipeline-operator.ts:48]: nit: name seems to be little long, can we simply keep it to `isTriggersGAVersion`

File: frontend/packages/pipelines-plugin/src/models/pipelines.ts [hunk 3]
```diff
@@ -145,7 +145,6 @@ export const TriggerBindingModel: K8sKind = {
   id: 'triggerbinding',
   labelPlural: 'TriggerBindings',
   crd: true,
-  badge: BadgeType.TECH,
```
Comment (abhinandan13jan) [inline frontend/packages/pipelines-plugin/src/models/pipelines.ts:148]: should we keep this for any future use?

File: frontend/packages/pipelines-plugin/src/utils/__tests__/hooks.spec.ts [hunk 4]
```diff
@@ -22,3 +22,23 @@ describe('usePipelineTechPreviewBadge:', () => {
     });
   });
 });
+
+describe('useTriggersTechPreviewBadge:', () => {
+  it('should return the badge if triggers GA operator is not installed', () => {
+    jest.spyOn(operatorUtils, 'usePipelineOperatorVersion').mockReturnValue({ version: '1.5.2' });
+    testHook(() => {
+      const badge = useTriggersTechPreviewBadge('test-ns');
+      expect(badge).not.toBeNull();
```
Comment (abhinandan13jan) [inline frontend/packages/pipelines-plugin/src/utils/__tests__/hooks.spec.ts:31]: can we have a better test than null? like checking <TechPreviewBadge>

**Comment by abhinandan13jan** [comment]
PR works fine. tested locally. Only coudn't test the Trigger Template page for the spec error on 1.6.

File: frontend/packages/pipelines-plugin/src/models/pipelines.ts (hunk 3 above)
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/models/pipelines.ts:148]: Moving forward, we no longer need this badges for both pipeline and trigger components.

**Pull request merged**